crates/ide-completion/src/context/analysis.rs

```diff
@@ -1129,7 +1129,22 @@ fn classify_name_ref(
         let is_trailing_outer_attr = kind != AttrKind::Inner
             && non_trivia_sibling(attr.syntax().clone().into(), syntax::Direction::Next).is_none();
         let annotated_item_kind = if is_trailing_outer_attr { None } else { Some(attached.kind()) };
-        Some(PathKind::Attr { attr_ctx: AttrCtx { kind, annotated_item_kind } })
+        let derive_helpers = annotated_item_kind
+            .filter(|kind| {
+                matches!(
+                    kind,
+                    SyntaxKind::STRUCT
+                        | SyntaxKind::ENUM
+                        | SyntaxKind::UNION
+                        | SyntaxKind::VARIANT
+                        | SyntaxKind::TUPLE_FIELD
+                        | SyntaxKind::RECORD_FIELD
+                )
+            })
```

Comment:
These could be used within the type specifications within ADTs as well, as in `struct T { field: fn(#[helper] ()) }` if I am not mistaken, though this is good enough

Reply:
I just checked and that's not true.

Reply:
![image](https://github.com/user-attachments/assets/c98ff6e1-1d9c-40af-825c-a756b7534ccb)
It does work, but no one ever does this. rustfmt even drops the attribute in the function pointer param. So either way this is good enough

Reply:
Ah I thought you mean the opposite - that the attribute is not a helper but completed as such.